modules/blockexplorer/update.go

```diff
@@ -1,23 +1,61 @@
 package blockexplorer
 
 import (
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/encoding"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-// Handles updates recieved from the consensus subscription, currently
-// just removing and inserting blocks from the blockchain
+// Handles updates recieved from the consensus subscription. Keeps
+// track of transaction volume, block timestamps and block sizes, as
+// well as the current block height
 func (be *BlockExplorer) ReceiveConsensusSetUpdate(cc modules.ConsensusChange) {
-	// Do lock stuff here
 	lockID := be.mu.Lock()
 	defer be.mu.Unlock(lockID)
 
-	// Update the current block, and block height
+	// Modify the currency value
+	for _, diff := range cc.SiacoinOutputDiffs {
+		if diff.Direction == true {
+			be.currencySent = be.currencySent.Add(diff.SiacoinOutput.Value)
+		}
+	}
+
+	// Modify the number of file contracts and their values
+	for _, diff := range cc.FileContractDiffs {
+		if diff.Direction == true {
+			be.fileContracts += 1
+			be.fileContractCost = be.fileContractCost.Add(diff.FileContract.Payout)
+		} else {
+			be.fileContracts -= 1
+			be.fileContractCost = be.fileContractCost.Sub(diff.FileContract.Payout)
+		}
```

Comment:
it would be interesting to have both a 'currently open' and an 'ever open' value.